2021/classes/submarine.py

```diff
@@ -0,0 +1,22 @@
+from turtle import numinput
+
+
+class Submarine:
+    def __init__(self, readings):
+        self.readings = readings
+        self.simple_increases = 0
+        self.deep_increases = 0
+
+    @staticmethod
+    def sonar_sweep(readings):
+        return len([True for r1, r2 in zip(readings, readings[1:]) if r2 > r1])
+
+    def simple_sonar_sweep(self):
+        self.simple_increases = self.sonar_sweep(self.readings)
+
+    def deep_sonar_sweep(self):
+        triple_step_sums = [
+            sum([r1, r2, r3])
+            for r1, r2, r3 in zip(self.readings, self.readings[1:], self.readings[2:])
+        ]
+        self.deep_increases = self.sonar_sweep(triple_step_sums)
```

Comment:
I don't love the API of these methods setting an instance variable instead of directly returning the value.

This API requires you to invoke a method, and then know which instance variable to check to retrieve the result, which does not lend itself to being very discoverable.

I would much rather the methods return the value directly. Writing tests for these methods would likely help surface this issue, as you'd more directly feel the pain of invoking methods, and asserting against a side effect that occurs elsewhere.

Think about if you were to write a doc for these methods. Would it feel good to need to explain that there are extra steps to retrieve the result? Probably not!